src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,86 @@
+import { CSSProperties, KeyboardEvent } from "react";
+import { Close } from "@material-ui/icons";
+import styled from "styled-components";
+import _IconButton from "../IconButton";
+
+interface DialogProps {
+  open: boolean;
+  disableCloseButton?: boolean;
+  onClose?: () => void;
+  children?: React.ReactNode;
+  className?: string;
+  style?: CSSProperties;
+}
+
+export default function Dialog({
+  open,
+  disableCloseButton = false,
```

Comment:
[nitpick] - I think you won't need to set `false` if its type is `boolean` since optional(undefined) is also falsy.